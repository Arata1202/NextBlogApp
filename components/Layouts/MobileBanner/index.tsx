'use client';
import { useTheme } from 'next-themes';
export default function MobileBanner() {
  const { theme } = useTheme();
  return (
    <div className={`flex justify-center mt-20 ${theme === 'dark' ? 'bg-gray-500' : 'd9d9d9'}`}>
      <div style={{ position: 'relative', width: '700px' }}>
        <img
          loading="lazy"
          style={{ padding: '0 12px' }}
          src={
            theme === 'dark'
              ? '/images/blog/mobilebanner-dark.png'
              : '/images/blog/mobilebanner.png'
          }
          alt=""
        />
        <div
          style={{
            position: 'absolute',
            bottom: '10px',
            left: '15px',
            display: 'flex',
            justifyContent: 'center',
            width: '45%',
          }}
        >
          <a
            href="https://apps.apple.com/jp/app/リアル大学生-モバイル/id6590619103"
            target="_blank"
            className="hover:opacity-60"
          >
            <img
              src="/images/badge/Download_on_the_App_Store_Badge_US-UK_RGB_blk_092917.svg"
              alt="Download on the App Store"
              style={{ height: '40px' }}
              className="app-badge"
            />
          </a>
          <a
            href="https://play.google.com/store/apps/details?id=com.realunivlog.flutterblogapp"
            target="_blank"
            className="app-badge-link hover:opacity-60"
            style={{ marginLeft: '10%' }}
          >
            <img
              src="/images/badge/GetItOnGooglePlay_Badge_Web_color_English.png"
              alt="Download on the Play Store"
              style={{ height: '40px' }}
              className="app-badge"
            />
          </a>
        </div>
      </div>
    </div>
  );
}
