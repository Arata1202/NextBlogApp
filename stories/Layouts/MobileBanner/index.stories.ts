import type { Meta, StoryObj } from '@storybook/react';
import MobileBanner from '@/components/Layouts/MobileBanner';

const meta: Meta<typeof MobileBanner> = {
  title: 'Layouts/MobileBanner',
  component: MobileBanner,
  parameters: {
    layout: 'fullscreen',
  },
  argTypes: {
    isDarkMode: { control: 'boolean' },
  },
};

export default meta;
type Story = StoryObj<typeof meta>;

export const Default: Story = {
  args: {
    isDarkMode: false,
  },
};
