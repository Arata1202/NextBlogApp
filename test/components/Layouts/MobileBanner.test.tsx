import { render } from '@testing-library/react';
import MobileBanner from '@/components/Layouts/MobileBanner';

describe('MobileBanner', () => {
  test('スナップショット（MobileBanner）', () => {
    const { asFragment } = render(<MobileBanner />);
    expect(asFragment()).toMatchSnapshot();
  });
});
