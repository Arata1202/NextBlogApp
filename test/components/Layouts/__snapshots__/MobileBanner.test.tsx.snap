// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`MobileBanner スナップショット（MobileBanner） 1`] = `
<DocumentFragment>
  <div
    class="flex justify-center mt-20"
    style="background-color: rgb(217, 217, 217);"
  >
    <div
      style="position: relative; width: 700px;"
    >
      <img
        alt=""
        loading="lazy"
        src="/images/blog/mobilebanner.png"
        style="padding: 0px 12px;"
      />
      <div
        style="position: absolute; bottom: 10px; left: 15px; display: flex; justify-content: center; width: 45%;"
      >
        <a
          class="hover:shadow-xl transition-shadow duration-200 transform hover:-translate-y-1"
          href="https://apps.apple.com/jp/app/リアル大学生-モバイル/id6590619103"
          target="_blank"
        >
          <img
            alt="Download on the App Store"
            class="app-badge"
            src="/Download_on_the_App_Store_Badge_US-UK_RGB_blk_092917.svg"
            style="height: 40px;"
          />
        </a>
        <a
          class="app-badge-link hover:shadow-xl transition-shadow duration-200 transform hover:-translate-y-1"
          href="https://play.google.com/store/apps/details?id=com.realunivlog.flutterblogapp"
          style="margin-left: 10%;"
          target="_blank"
        >
          <img
            alt="Download on the Play Store"
            class="app-badge"
            src="/GetItOnGooglePlay_Badge_Web_color_English.png"
            style="height: 40px;"
          />
        </a>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
